### PR TITLE
Added rules for vivado. These enable a bitstream to be built.

### PR DESCRIPTION
**Pull request by stridge-cruxml**
Manually built the bitstream in the test. Discussed in #106.

Thanks @rickwebiii for [bazel_fpga_rules](https://github.com/rickwebiii/bazel_fpga_rules)
Thanks @agoessling for [rules_vivado](https://github.com/agoessling/rules_vivado)

I used both of them as a reference and added some of my own ideas.

### CODE REVIEW

**Comment by stridge-cruxml** [comment]
@QuantamHD ready to merge if you think it's ok?

**Comment by mithro** [comment]
FYI - @QuantamHD is out of office until next Monday.

**Comment by stridge-cruxml** [comment]
I added support for xsim tests and for Vitis/Vivado HLS [here](https://github.com/CruxML/bazel_rules_hdl/tree/feature/hls).

Separate PRs or i just merge them into this one too?

**Comment by QuantamHD** [comment]
Let's do separate PRs

File: vivado/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+It has the optional argument `_create_project_tcl_template` which by default loads [create_project.tcl.template](create_project.tcl.template). This can be modified.
+
+## synthesize
+
+Creates a synthesized checkpoint.
+
+Example from `vivado/tests`:
+```
+synthesize(
```
[the remark this answers is not in the document]
Comment (stridge-cruxml) [reply vivado/README.md:70]: I had a look at the existing flow stuff and it wasn't clear to me how to go about it or if it should be done. I will probably leave that to others.

Added the prefix `vivado_`.

File: vivado/defs.bzl [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+        "{{HDL_SOURCE_CONTENT}}": hdl_source_content,
+        "{{TCL_CONTENT}}": tcl_content,
+        "{{CONSTRAINTS_CONTENT}}": constraints_content,
+        "{{MODULE_TOP}}": ctx.attr.module_top,
+        "{{PROJECT_DIR}}": project_dir.path,
+        "{{SYNTH_STRATEGY}}": synth_strategy_str,
+        "{{SYNTH_CHECKPOINT}}": synth_path,
+        "{{TIMING_SUMMARY_REPORT}}": timing_path,
+        "{{UTILIZATION_REPORT}}": util_path,
+        "{{JOBS}}": "4",
```
[the remark this answers is not in the document]
Comment (stridge-cruxml) [reply vivado/defs.bzl]: Done.

File: vivado/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ted-xie) [reply vivado/README.md:70]: Did you end up updating this README to reflect the `vivado_` prefix?

FWIW, I think it also would've been fine to call the rule `synthesize`. Users could organize their projects with `asic` and `xilinx` subdirectories, and run `bazel build //cad/asic:synth //cad/xilinx:synth` for example.

**Pull request merged**